document/src/vespa/document/fieldvalue/boolfieldvalue.h

```diff
@@ -41,6 +41,12 @@ class BoolFieldValue : public FieldValue {
     double getAsDouble() const override;
     vespalib::string getAsString() const override;
 
+    FieldValue& operator=(vespalib::stringref) override;
+    FieldValue& operator=(int32_t) override;
+    FieldValue& operator=(int64_t) override;
+    FieldValue& operator=(float) override;
+    FieldValue& operator=(double) override;
```

Comment:
Could consider returning `BoolFieldValue&` here since reference return types are covariant

Reply:
Fixed. And yes this is scary stuff.